### PR TITLE
feat(gatsby-theme-bodiless, core, ): Add Metadata with GIT Values (Bodiless) #1331

### PR DESCRIPTION
**Pull request by dewen**
## Related Issues

Fixes #1331

### CODE REVIEW

File: packages/bodiless-components/src/Meta/Meta.tsx [hunk 1]
```diff
@@ -106,6 +107,17 @@ const withMetaHtml = (
   );
 };
 
+const withMetaSiteInfo = (HelmetComponent: CT) => (props: any) => {
+  const { children, ...rest } = props;
+  const siteInfo = btoa(JSON.stringify(useGitContext()));
```
Comment (dewen) [inline packages/bodiless-components/src/Meta/Meta.tsx:112]: btoa creates base64 encoded string on client side. This obfuscate of data could be moved to backend.

File: packages/bodiless-core/src/Git/GitContextProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+const GitContext = React.createContext<GitContextProps>(defaultGitInfo);
+
+/**
+ * Component which provides its children with `GitContext` of site git info.
+ *
+ * @param props
+ */
+export const GitContextProvider: FC<GitContextProviderProps> = ({ children, ...rest }) => {
+  const { gitInfo } = rest;
```
Comment (lucas-varela) [inline packages/bodiless-core/src/Git/GitContextProvider.tsx:32]: This destructuring is redundant.

```suggestion
export const GitContextProvider: FC<GitContextProviderProps> = ({ children, gitInfo }) => {
```

File: packages/gatsby-theme-bodiless/create-node.js [hunk 3]
```diff
@@ -61,6 +66,30 @@ const findFilesystemNode = ({ node, getNode }) => {
...
+ *  hash: string,
+ *  branch: string,
+ * }>
+ */
+const createGitInfo = async () => {
+  const gitDir = await findGitFolder();
+  const projectRoot = pathUtil.dirname(gitDir);
+  const remotes = await git.listRemotes({ fs, dir: projectRoot });
+  const origin = remotes.filter(v => v.remote === 'origin');
+  const repo = (origin.length === 1) ? origin[0].url : '';
```
Comment (lucas-varela) [inline packages/gatsby-theme-bodiless/create-node.js:83]: Using `Array.find` would probably make these lines clearer:

```suggestion
  const origin = remotes.find(v => v.remote === 'origin');
  const repo = origin?.url ?? '';
```

File: packages/gatsby-theme-bodiless/create-node.js [hunk 4]
```diff
@@ -61,6 +66,30 @@ const findFilesystemNode = ({ node, getNode }) => {
...
+ * Get current git repo info.
+ *
+ * @returns Promise<{
+ *  repo: string,
+ *  hash: string,
+ *  branch: string,
+ * }>
+ */
+const createGitInfo = async () => {
+  const gitDir = await findGitFolder();
```
Comment (lucas-varela) [inline packages/gatsby-theme-bodiless/create-node.js:79]: What happens if Bodiless is deployed without using git? For instance, a CI/CD system might generate a build and deploy it directly into a server, so the .git folder might not be included. `gitDir` could be undefined, or a path to a git repo not related to Bodiless found somewhere up in the file system.

Should this code account for these cases?

File: packages/gatsby-theme-bodiless/create-node.js [hunk 5]
```diff
@@ -18,9 +18,12 @@
 const pathUtil = require('path');
 const slash = require('slash');
 const crypto = require('crypto');
-const fs = require('fs-extra');
+const fs = require('fs');
```
Comment (lucas-varela) [inline packages/gatsby-theme-bodiless/create-node.js:21]: This import could be removed as it's not used anymore.

File: sites/test-site/src/templates/product_listing.jsx [hunk 6]
```diff
@@ -24,8 +26,11 @@ import {
 } from '../components/ProductListing';
 import { FilterByGroupSingleSiteWide } from '../components/FilterByGroup';
 
+const SiteInfoHelmet = withMetaSiteInfo(Helmet);
+
 const main = props => (
   <Page {...props}>
+    <SiteInfoHelmet />
```
Comment (lucas-varela) [inline sites/test-site/src/templates/product_listing.jsx:33]: Is this the only page where site info metadata will be available? I thought all pages would have to include this data.

**Review by lucas-varela — CHANGES_REQUESTED** [review]

File: packages/gatsby-theme-bodiless/create-node.js (hunk 5 above)
Comment (dewen) [reply packages/gatsby-theme-bodiless/create-node.js:21]: @lucas-varela actually the `fs` is required by `listRemotes` on line 81.

I renamed `fs-extra` to `fse` in order to introduce `fs` here.

File: packages/gatsby-theme-bodiless/create-node.js (hunk 4 above)
Comment (dewen) [reply packages/gatsby-theme-bodiless/create-node.js:79]: updated

**Review by lucas-varela — APPROVED** [review]

File: sites/test-site/src/templates/product_listing.jsx (hunk 6 above)
Comment (dewen) [reply sites/test-site/src/templates/product_listing.jsx:33]: moved to default template page according to AC1

**Review by lucas-varela — APPROVED** [review]

**Review by irkhodzka — APPROVED** [review]

**Pull request merged**